src/modules/peek/Peek.UI/FileQuery.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Peek.UI
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using CommunityToolkit.Mvvm.ComponentModel;
+    using Peek.Common.Models;
+    using Peek.UI.Helpers;
+
+    public partial class FileQuery : ObservableObject
+    {
+        private const int UninitializedItemIndex = -1;
+
+        public void Uninitialize()
+        {
+            CurrentFile = null;
+
+            if (initializeFilesTask != null && initializeFilesTask.Status == TaskStatus.Running)
+            {
+                Debug.WriteLine("Detected existing initializeFilesTask running. Cancelling it..");
+                cancellationTokenSource.Cancel();
+            }
+
+            initializeFilesTask = null;
+
+            lock (mutateQueryDataLock)
+            {
+                files = new List<File>();
+                currentItemIndex = UninitializedItemIndex;
+            }
+        }
+
+        public void UpdateCurrentItemIndex(int desiredIndex)
```

Comment:
nit: naming: UpdateCurrentFileIndex?  Should we just use the term file? Item could be confusing.

Reply:
I'm gonna try to remove any mention of 'file' in my variable names, since we're these items could folders too